From d1aa6e5f036590a5cfe119545ac4af3f3dc8424f Mon Sep 17 00:00:00 2001
From: Nilima Paul <nilimapaul002@gmail.com>
Date: Thu, 22 Jul 2021 15:16:32 +0530
Subject: [PATCH] Create Patch File

---
 client3.py | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/client3.py b/client3.py
index f1771c3..5905445 100644
--- a/client3.py
+++ b/client3.py
@@ -35,14 +35,11 @@ def getDataPoint(quote):
 	stock = quote['stock']
 	bid_price = float(quote['top_bid']['price'])
 	ask_price = float(quote['top_ask']['price'])
-	price = bid_price
+	price = (bid_price + ask_price)/2
 	return stock, bid_price, ask_price, price
 
 def getRatio(price_a, price_b):
-	""" Get ratio of price_a and price_b """
-	""" ------------- Update this function ------------- """
-	""" Also create some unit tests for this function in client_test.py """
-	return 1
+		return price_a/price_b
 
 # Main
 if __name__ == "__main__":
@@ -52,8 +49,12 @@ if __name__ == "__main__":
 		quotes = json.loads(urllib.request.urlopen(QUERY.format(random.random())).read())
 
 		""" ----------- Update to get the ratio --------------- """
+		# prices dictionery holds the key as stock name and value as price of stock
+		prices = {}
+
 		for quote in quotes:
 			stock, bid_price, ask_price, price = getDataPoint(quote)
-			print ("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
+			prices[stock] = price
+			print("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 
-		print ("Ratio %s" % getRatio(price, price))
+		print("Ratio %s" % getRatio(prices['ABC'], prices['DEF']))
\ No newline at end of file
-- 
2.31.1.windows.1

